libs/language-server/src/lib/builtin-library/stdlib.ts

```diff
@@ -2,21 +2,117 @@
 //
 // SPDX-License-Identifier: AGPL-3.0-only
 
+import { IOType } from '../ast';
 import { PrimitiveValuetypes } from '../ast/wrappers/value-type/primitive/primitive-valuetypes';
+import {
+  BlockMetaInformation,
+  metaInformationRegistry,
+} from '../meta-information';
 
 import { PartialStdLib } from './generated/partial-stdlib';
 
-export const BuiltinValuetypesLib = {
-  'builtin:///stdlib/builtin-valuetypes.jv': Object.values(PrimitiveValuetypes)
-    .filter((v) => v.isUserExtendable())
-    .map(
-      (valueType) =>
-        `${(valueType.getUserDoc()?.trim().split('\n') ?? [])
-          .map((t) => '// ' + t)
-          .join('\n')}
+export function getBuiltinValuetypesLib() {
+  const primitiveValuetypes = Object.values(PrimitiveValuetypes).map(
+    (valueType) =>
+      `${(valueType.getUserDoc()?.trim().split('\n') ?? [])
+        .map((t) => '// ' + t)
+        .join('\n')}
 builtin valuetype ${valueType.getName()};`,
-    )
+  );
+
+  const collectionValuetype = `builtin valuetype Collection<ElementType>;`;
+
+  return {
+    'builtin:///stdlib/builtin-valuetypes.jv': [
+      ...primitiveValuetypes,
+      collectionValuetype,
+    ].join('\n\n'),
+  };
+}
+
+export const IOtypesLib = {
+  'builtin:///stdlib/iotypes.jv': Object.values(IOType)
+    .map((iotype) => `builtin iotype ${iotype};`)
     .join('\n\n'),
 };
 
-export const StdLib = { ...PartialStdLib, ...BuiltinValuetypesLib };
+// Is a method since metaInformationRegistry might not be initialized when this as variable.
+export function getBulitinBlocktypesLib() {
+  return {
+    'builtin:///stdlib/builtin-blocktypes.jv': metaInformationRegistry
+      .getAllEntries()
+      .reduce(
```

Comment:
filter -> map would be easier to read, also consider ".filter(Boolean)" to filter out undefined values.